### PR TITLE
engines: Abort each run

### PR DESCRIPTION
**Pull request by akihikodaki**
Not tested.

### CODE REVIEW

File: lib/setupmanagers/qemuhck/qemu_machine.rb [hunk 1]
```diff
@@ -54,8 +54,6 @@ def initialize(options)
 
     def define_local_variables
       @devices_list = []
-      @pre_start_commands = []
```
Comment (kostyanf14) [inline lib/setupmanagers/qemuhck/qemu_machine.rb:57]: Why did you remove this? We have some devices such as ivshmem/virtio-fs where we MUST run the daemon on the host before tests and kill it after it.

**Review by kostyanf14 — COMMENTED** [review]
I am testing these patches. I will comment more later.

File: lib/setupmanagers/qemuhck/qemu_machine.rb (hunk 1 above)
Comment (akihikodaki) [reply lib/setupmanagers/qemuhck/qemu_machine.rb]: I missed their usage. Commit e82eedb3dfaef22b24a28a4972a0bfe6b4dd8938 restores them.

Comment (kostyanf14) [reply lib/setupmanagers/qemuhck/qemu_machine.rb]: For example: https://github.com/HCK-CI/AutoHCK/blob/master/lib/setupmanagers/qemuhck/devices/ivshmem-doorbell.json

**Review by YanVugenfirer — APPROVED** [review]
